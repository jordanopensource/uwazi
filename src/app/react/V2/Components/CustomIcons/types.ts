interface CustomIconProps {
  className?: string;
  color?: string;
}

export type { CustomIconProps };
