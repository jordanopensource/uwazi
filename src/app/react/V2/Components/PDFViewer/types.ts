import { TextSelection } from '@huridocs/react-text-selection-handler/dist/TextSelection';

type TextHighlight = {
  key: string;
  textSelection: TextSelection;
  color?: string;
};

export type { TextHighlight };
