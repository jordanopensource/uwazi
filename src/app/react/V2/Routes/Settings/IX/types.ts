import { IXExtractorInfo } from 'V2/shared/types';
import { ObjectIdSchema, PropertySchema } from 'shared/types/commonTypes';
import { TextHighlight } from 'V2/Components/PDFViewer/types';
import { EntitySuggestionType } from 'shared/types/suggestionType';

interface IXProperty extends PropertySchema {
  type: 'text' | 'date' | 'numeric' | 'markdown';
}

type Extractor = IXExtractorInfo & {
  rowId: string;
  namedTemplates: string[];
  propertyLabel: string;
  propertyType: IXProperty['type'];
};

type Highlights = { [page: string]: TextHighlight[] };

type SuggestionValue = string | number;

interface SingleValueSuggestion extends EntitySuggestionType {
  _id: ObjectIdSchema;
  rowId: string;
  disableRowSelection?: boolean;
  suggestedValue: SuggestionValue;
  currentValue?: SuggestionValue;
  isChild?: boolean;
}

interface MultiValueSuggestion extends EntitySuggestionType {
  _id: ObjectIdSchema;
  rowId: string;
  disableRowSelection?: boolean;
  suggestedValue: SuggestionValue[];
  currentValue: SuggestionValue[];
  subRows: SingleValueSuggestion[];
  isChild?: boolean;
}

type TableSuggestion = SingleValueSuggestion | MultiValueSuggestion;

type TableExtractor = Extractor & { rowId: string };

export type {
  Extractor,
  TableExtractor,
  Highlights,
  TableSuggestion,
  SuggestionValue,
  SingleValueSuggestion,
  MultiValueSuggestion,
};
