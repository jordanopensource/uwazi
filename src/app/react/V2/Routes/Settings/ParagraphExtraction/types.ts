export type Extractor = {
  _id: string;
  templateFrom: string[];
  templateTo: string;
  documents: number;
  generatedEntities: number;
};

export type TableExtractor = Extractor & {
  rowId: string;
  targetTemplateName: string;
  originTemplateNames: string[];
};
