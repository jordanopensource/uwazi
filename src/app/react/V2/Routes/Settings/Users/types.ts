import { ClientUserGroupSchema, ClientUserSchema } from 'app/apiResponseTypes';

type FormIntent =
  | 'new-user'
  | 'edit-user'
  | 'delete-users'
  | 'new-group'
  | 'edit-group'
  | 'delete-groups'
  | 'unlock-user'
  | 'reset-password'
  | 'reset-2fa'
  | 'bulk-reset-2fa'
  | 'bulk-reset-password';

type User = ClientUserSchema & { rowId: string };
type Group = ClientUserGroupSchema & { rowId: string };

export type { FormIntent, User, Group };
