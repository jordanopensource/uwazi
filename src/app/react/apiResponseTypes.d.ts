import { UserGroupSchema } from 'shared/types/userGroupType';
import { UserSchema } from 'shared/types/userType';
import {
  PreserveConfig,
  Settings,
  SettingsFilterSchema,
  SettingsLinkSchema,
} from 'shared/types/settingsType';
import { LanguageSchema } from 'shared/types/commonTypes';
import { ThesaurusSchema, ThesaurusValueSchema } from 'shared/types/thesaurusType';
import { TemplateSchema } from 'shared/types/templateType';

export interface GroupMemberSchema {
  refId: string;
  username: string;
}

export interface ClientUserGroupSchema extends Omit<UserGroupSchema, '_id' | 'members'> {
  _id?: string;
  members: GroupMemberSchema[];
}

export interface ClientUserSchema extends Omit<UserSchema, '_id' | 'groups'> {
  _id?: string;
  groups?: {
    _id: string;
    name: string;
  }[];
}

export interface ClientSettingsFilterSchema extends Omit<SettingsFilterSchema, '_id'> {
  _id?: string;
}

export interface ClientSublink extends Omit<ClientSettingsLinkSchema, 'sublinks'> {}

export interface ClientSettingsLinkSchema extends Omit<SettingsLinkSchema, '_id' | 'sublinks'> {
  _id?: string;
  sublinks?: ClientSublink[];
}

export interface ClientPreserveConfig extends Omit<PreserveConfig, 'config'> {
  config: {
    token: string;
    template: string;
    user?: string;
  }[];
}

export interface ClientLanguageSchema extends Omit<LanguageSchema, '_id'> {
  _id?: string;
}

export interface Template extends TemplateSchema, Omit<TemplateSchema, '_id'> {
  _id: string;
}

export interface ClientSettings
  extends Omit<Settings, '_id | filters | links | features | languages'> {
  _id?: string;
  filters?: ClientSettingsFilterSchema[];
  languages?: ClientLanguageSchema[];
  links?: ClientSettingsLinkSchema[];
  features?: Omit<Settings['features'], 'preserve | metadataExtraction'> & {
    preserve?: ClientPreserveConfig;
    ocr?: { url: string };
    metadataExtraction?: {
      url: string;
      templates?: {
        template: string;
        properties: string[];
      }[];
    };
    paragraphExtraction?: boolean;
    [k: string]: unknown | undefined;
  };
}

export interface ClientThesaurus extends ThesaurusSchema, Omit<ThesaurusSchema, '_id | values'> {
  _id: ObjectIdSchema;
  values: ClientThesaurusValue[];
}

export interface ClientThesaurusValue extends Omit<ThesaurusValueSchema, '_id'> {
  id?: string;
  label: string;
  values?: {
    id?: string;
    label: string;
    name?: string;
  }[];
}

export interface ClientRelationshipType {
  _id: string;
  name: string;
}
