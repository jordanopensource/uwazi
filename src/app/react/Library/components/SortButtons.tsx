import React from 'react';
import { bindActionCreators, Dispatch } from 'redux';
import { connect, ConnectedProps } from 'react-redux';
import { actions } from 'react-redux-form';
import { wrapDispatch } from 'app/Multireducer';
import { Icon } from 'UI';
import { DropdownList } from 'app/Forms';
import { IImmutable } from 'shared/types/Immutable';
import { IStore } from 'app/istore';
import { omit } from 'lodash';
import {
  filterTemplates,
  getPropertySortType,
  getSortOptions,
  SearchOptions,
  SortType,
} from '../helpers/sortComponets';

interface SortButtonsOwnProps {
  stateProperty: string;
  selectedTemplates: IImmutable<string[]>;
  sortCallback: Function;
}

const mapStateToProps = (state: IStore, ownProps: SortButtonsOwnProps) => {
  let { templates } = state;
  const stateProperty = ownProps.stateProperty ? ownProps.stateProperty : 'library.search';

  if (ownProps.selectedTemplates && ownProps.selectedTemplates.count()) {
    templates = filterTemplates(state.templates, ownProps.selectedTemplates);
  }
  const search = stateProperty
    .split(/[.,/]/)
    .reduce(
      (memo: { [k: string]: any }, property: string) =>
        Object.keys(memo).indexOf(property) !== -1 ? memo[property] : null,
      state
    );
  return { ...ownProps, stateProperty, search, templates };
};

const mapDispatchToProps = (dispatch: Dispatch<{}>) =>
  bindActionCreators({ merge: actions.merge }, wrapDispatch(dispatch, 'library'));

const connector = connect(mapStateToProps, mapDispatchToProps);

type mappedProps = ConnectedProps<typeof connector>;

const getToggleSearchIcon = (search: SearchOptions) =>
  search.order === 'asc' && search.sort !== '_score' ? 'arrow-up' : 'arrow-down';

const validateSearch = (search: SearchOptions): SearchOptions =>
  search.sort === '_score' && !search.searchTerm
    ? {
        sort: 'creationDate',
        order: 'desc',
        searchTerm: search.searchTerm,
      }
    : { searchTerm: search.searchTerm, sort: search.sort };

const sortDirection = (condition: string) => {
  switch (condition) {
    case 'desc':
    case 'string':
      return 'asc';
    case 'asc':
    default:
      return 'desc';
  }
};

const SortButtonsComponent = ({
  search,
  merge,
  stateProperty,
  templates,
  sortCallback,
}: mappedProps) => {
  const doSort = (property: string, defaultTreatAs: string, selectedSort?: string) => {
    const treatAs = defaultTreatAs;
    const order = selectedSort || sortDirection(treatAs);
    const newSort = { sort: property, order, treatAs };
    merge(stateProperty, newSort);
    const filters = { ...search, ...omit(newSort, 'treatAs'), userSelectedSorting: true };
    return sortCallback && sortCallback({ search: filters }, 'library');
  };

  const changeOrder = () => {
    doSort(search.sort, search.treatAs, sortDirection(search.order));
  };

  const validatedSearch = validateSearch(search);

  const sortOptions = getSortOptions(search, templates);

  return (
    <div className="sort-buttons">
      <DropdownList
        className="sort-dropdown"
        value={validatedSearch.sort}
        data={sortOptions}
        valueField="value"
        textField="label"
        onChange={(selected: SortType) => doSort(selected.value, getPropertySortType(selected))}
      />
      <button
        type="button"
        disabled={search.sort === '_score'}
        className={`sorting-toggle ${search.sort === '_score' && 'disabled'}`}
        onClick={changeOrder}
      >
        <Icon icon={getToggleSearchIcon(search)} />
      </button>
    </div>
  );
};

const container = connector(SortButtonsComponent);
export type { SortButtonsOwnProps };
export { container as SortButtons, mapStateToProps };
