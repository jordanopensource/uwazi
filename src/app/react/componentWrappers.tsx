/* eslint-disable comma-spacing */
/* eslint-disable react/no-multi-comp */
import React, { useContext } from 'react';
import {
  useLocation,
  useMatches,
  useNavigate,
  useOutlet,
  useParams,
  useSearchParams,
} from 'react-router-dom';
import { AppMainContext } from './App/AppMainContext';
import { searchParamsFromSearchParams } from './utils/routeHelpers';

const withRouter =
  <T,>(Component: React.ComponentClass<T & { params?: any }, any>) =>
  (props: T & { params?: any }) => {
    const location = useLocation();
    const navigate = useNavigate();
    const params = useParams();
    const matches = useMatches();
    const [searchParams, setSearchParams] = useSearchParams();
    const { q: searchString } = searchParamsFromSearchParams(searchParams);
    const paramsFromProps = { ...props.params, q: searchString || props.params?.q };

    return (
      <Component
        {...props}
        navigate={navigate}
        location={location}
        params={{ ...params, ...(paramsFromProps || {}) }}
        matches={matches}
        searchParams={searchParams}
        setSearchParams={setSearchParams}
      />
    );
  };

const withContext =
  <T,>(Component: React.ComponentClass<T, any>) =>
  (props: T) => {
    const mainContext = useContext(AppMainContext);
    return <Component {...props} mainContext={mainContext} />;
  };

const withOutlet =
  <T,>(Component: React.ComponentClass<T, any>) =>
  (props: T) => {
    const outlet = useOutlet(AppMainContext);
    return <Component {...props} outlet={outlet} />;
  };

export { withRouter, withContext, withOutlet };
