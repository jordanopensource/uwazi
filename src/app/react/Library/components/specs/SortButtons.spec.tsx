/**
 * @jest-environment jsdom
 */
import React from 'react';
import { fireEvent, RenderResult, screen } from '@testing-library/react';

import { defaultState, renderConnectedContainer } from 'app/utils/test/renderConnected';
import { fromJS, fromJS as immutable } from 'immutable';

import { IStore } from 'app/istore';
import { actions } from 'react-redux-form';
import { IImmutable } from 'shared/types/Immutable';
import { SortButtons, SortButtonsOwnProps, mapStateToProps } from '../SortButtons';

describe('SortButtons', () => {
  let props: SortButtonsOwnProps;
  let state: Partial<Omit<IStore, 'library'>> & {
    library: { search: { searchTerm?: string; sort?: string; order?: string; treatAs?: string } };
  };
  let renderResult: RenderResult;

  const render = () => {
    ({ renderResult } = renderConnectedContainer(<SortButtons {...props} />, () => state));
  };

  const openDropdown = () => {
    const openBtn = renderResult.container.getElementsByClassName('rw-btn-select')[0];
    fireEvent.click(openBtn);
    const options = screen.getAllByRole('option');
    const optionsLabel = options.map(option => option.textContent);
    return { options, optionsLabel };
  };
  const selectOption = (options: Element[], label: string) => {
    const optionToSelect = options.find(option => option.textContent === label);
    fireEvent.click(optionToSelect!);
  };
  beforeEach(() => {
    state = {
      ...defaultState,
      templates: immutable([
        {
          _id: 'id',
          properties: [
            {},
            { filter: true, name: 'date', label: 'date', type: 'date' },
            { filter: true, name: 'number', label: 'number', type: 'numeric' },
            { filter: true, name: 'my_select', label: 'my select', type: 'select' },
            { filter: true, name: 'sortable_name', label: 'sortableProperty', type: 'text' },
            {
              filter: true,
              name: 'inherited1',
              label: 'inherited1',
              type: 'relationship',
              inherit: {
                type: 'text',
              },
            },
            {
              filter: true,
              name: 'inherited2',
              label: 'inherited2',
              type: 'relationship',
              inherit: {
                type: 'multiselect',
              },
            },
          ],
        },
      ]),
      library: {
        search: {
          sort: 'asc',
        },
      },
    };
    props = {
      sortCallback: jasmine.createSpy('sortCallback'),
      stateProperty: 'library.search',
      selectedTemplates: fromJS([]),
    };
  });

  describe('Sort options', () => {
    it('should use templates sortable properties as options (with asc and desc for each)', () => {
      render();
      expect(renderResult.asFragment()).toMatchSnapshot();
    });
    describe('when multiple options have the same name', () => {
      it('should not duplicate the entry', () => {
        state.templates = immutable([
          {
            _id: 'id',
            properties: [
              {},
              {
                filter: true,
                name: 'sortable_name',
                label: 'sortableProperty',
                type: 'text',
                _id: '_id',
              },
            ],
          },
          {
            properties: [
              {
                filter: true,
                name: 'sortable_name',
                label: 'anotherLabel',
                type: 'text',
                _id: '_id',
              },
            ],
          },
        ]);
        render();
        const { options, optionsLabel } = openDropdown();
        expect(options.length).toBe(4);
        expect(optionsLabel).toEqual(['Title', 'Date added', 'Date modified', 'sortableProperty']);
      });
    });
    describe('when there is an active search term', () => {
      it('should display sort by search relevance option', () => {
        state.library.search.searchTerm = 'keyword';
        render();
        const { optionsLabel } = openDropdown();
        expect(optionsLabel).toContain('Search relevance');
      });

      it('should disable the sort button and show the sort down icon', () => {
        state.library.search.searchTerm = 'keyword';
        state.library.search.sort = '_score';
        render();
        const toggleBtn = renderResult.container.getElementsByClassName(
          'sorting-toggle'
        )[0] as HTMLElement;
        expect(toggleBtn.attributes.getNamedItem('disabled')).toBeDefined();
        expect(toggleBtn.children[0].attributes.getNamedItem('data-icon')?.value).toBe(
          'arrow-down'
        );
      });
    });
    describe('when relevance sorting is active and there is no search term', () => {
      it('should fall back to sorting by most recent creation date', () => {
        state.library.search.sort = '_score';
        state.library.search.searchTerm = '';
        render();
        const combobox = screen.getByRole('combobox');
        expect(combobox.textContent).toBe('Date added');
        expect(renderResult.asFragment()).toMatchSnapshot();
      });
    });

    describe('when active', () => {
      it('should set the option active', () => {
        state.library.search.sort = 'metadata.sortable_name';
        render();
        const combobox = screen.getByRole('combobox');
        expect(combobox.textContent).toBe('sortableProperty');
      });
    });
    describe('clicking an option', () => {
      it('should sort by that property with default order (asc for text and desc for date)', () => {
        render();
        const { options } = openDropdown();
        selectOption(options, 'sortableProperty');
        expect(props.sortCallback).toHaveBeenCalledWith(
          { search: { sort: 'metadata.sortable_name', order: 'asc', userSelectedSorting: true } },
          'library'
        );
        selectOption(options, 'date');
        expect(props.sortCallback).toHaveBeenCalledWith(
          { search: { sort: 'metadata.date', order: 'desc', userSelectedSorting: true } },
          'library'
        );
      });
    });
  });
  describe('sort', () => {
    it('should not fail if no sortCallback', () => {
      /*  @ts-ignore */
      delete props.sortCallback;
      render();
      let error;
      state.library.search.sort = 'title';
      try {
        const searchBtn = screen.getByRole('button');
        fireEvent.click(searchBtn);
      } catch (err) {
        error = err;
      }
      expect(error).toBeUndefined();
    });

    it('should change sorting order when using the order button', () => {
      state.library.search = { order: 'asc', sort: 'title' };
      render();
      const searchBtn = screen.getByRole('button');
      fireEvent.click(searchBtn);
      expect(props.sortCallback).toHaveBeenCalledWith(
        { search: { sort: 'title', order: 'desc', userSelectedSorting: true } },
        'library'
      );
    });
    describe('when changing property being sorted', () => {
      const rerenderSortButtons = () => {
        state.library.search = { order: 'desc', sort: 'title' };
        render();
        // @ts-ignore
        props.sortCallback.calls.reset();
      };
      it('should use default order', () => {
        rerenderSortButtons();
        fireEvent.click(screen.getByRole('button'));
        expect(props.sortCallback).toHaveBeenCalledWith(
          { search: { sort: 'title', order: 'asc', userSelectedSorting: true } },
          'library'
        );
        renderResult.unmount();
        rerenderSortButtons();
        const { options } = openDropdown();
        selectOption(options, 'Date added');
        fireEvent.click(screen.getByRole('button'));
        expect(props.sortCallback).toHaveBeenCalledWith(
          { search: { sort: 'creationDate', order: 'desc', userSelectedSorting: true } },
          'library'
        );
      });
    });

    describe('when changing order', () => {
      it('should keep the treatAs property', () => {
        jest.spyOn(actions, 'merge');
        state.library.search = { order: 'desc', sort: 'title', treatAs: 'number' };
        render();
        const { options } = openDropdown();
        selectOption(options, 'Title');
        fireEvent.click(screen.getByRole('button'));
        expect(actions.merge).toHaveBeenCalledWith('library.search', {
          sort: 'title',
          order: 'asc',
          treatAs: 'number',
        });
      });
    });
  });
  describe('when filtering number property', () => {
    it('should set value to number', () => {
      state.library.search = { order: 'asc', sort: 'number' };
      render();
      const combobox = screen.getByRole('combobox');
      expect(combobox.textContent).toBe('number');
    });
  });

  describe('mapStateToProps', () => {
    let templates: IImmutable<any>[];

    it('should send all templates from state', () => {
      const stateToMap = { templates: immutable(['item']), library: { search: {} } };
      const _props = { storeKey: 'library' };
      // @ts-ignore
      expect(mapStateToProps(stateToMap, _props).templates.get(0)).toBe('item');
    });
    it('should only send selectedTemplates if array passed in ownProps', () => {
      templates = immutable([{ _id: 'a' }, { _id: 'b' }]);
      const stateToMap = { templates, library: { search: {} } };
      const _props = { selectedTemplates: immutable(['b']), storeKey: 'library' };
      // @ts-ignore
      expect(mapStateToProps(stateToMap, _props).templates.getIn([0, '_id'])).toBe('b');
    });

    describe('search', () => {
      beforeEach(() => {
        templates = immutable([{ _id: 'a' }, { _id: 'b' }]);
      });

      it('should be selected from the state according to the store key', () => {
        const stateValues = { templates, library: { search: 'correct search' } };
        const _props = { storeKey: 'library' };
        // @ts-ignore
        expect(mapStateToProps(stateValues, _props).search).toBe('correct search');
      });
      it('should be selected from the state according to the stateProperty (if passed)', () => {
        let stateToMap = {
          templates,
          library: { search: 'incorrect search', sort: 'correct search' },
        };
        let _props = { storeKey: 'library', stateProperty: 'library.sort' };
        // @ts-ignore
        expect(mapStateToProps(stateToMap, _props).search).toBe('correct search');

        stateToMap = {
          templates,
          library: {
            search: 'incorrect search',
            sort: 'incorrect search',
            // @ts-ignore
            nested: { dashed: 'correct search' },
          },
        };
        _props = { storeKey: 'library', stateProperty: 'library/nested.dashed' };
        // @ts-ignore
        expect(mapStateToProps(stateToMap, _props).search).toBe('correct search');
      });
    });
  });
});
