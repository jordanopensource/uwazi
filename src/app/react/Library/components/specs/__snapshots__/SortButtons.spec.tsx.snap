// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortButtons Sort options should use templates sortable properties as options (with asc and desc for each) 1`] = `
<DocumentFragment>
  <div
    class="sort-buttons"
  >
    <div
      aria-busy="false"
      aria-disabled="false"
      aria-expanded="false"
      aria-haspopup="true"
      aria-live="polite"
      aria-owns="rw_1_listbox"
      aria-readonly="false"
      class="sort-dropdown rw-dropdown-list rw-widget"
      id="rw_1_input"
      role="combobox"
      tabindex="0"
    >
      <div
        class="rw-widget-input rw-widget-picker rw-widget-container"
      >
        <div
          class="rw-input rw-dropdown-list-input"
        >
          <input
            class="rw-dropdown-list-autofill rw-detect-autofill rw-sr"
            tabindex="-1"
            value="asc"
          />
          asc
        </div>
        <span
          class="rw-select"
        >
          <button
            aria-disabled="false"
            aria-hidden="true"
            aria-label="open dropdown"
            class="rw-btn rw-btn-select"
            role="presentational"
            tabindex="-1"
            title="open dropdown"
            type="button"
          >
            <span
              aria-hidden="true"
              class="rw-i rw-i-caret-down"
            />
          </button>
        </span>
      </div>
    </div>
    <button
      class="sorting-toggle false"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-down fa-w-14 "
        data-icon="arrow-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;

exports[`SortButtons Sort options when relevance sorting is active and there is no search term should fall back to sorting by most recent creation date 1`] = `
<DocumentFragment>
  <div
    class="sort-buttons"
  >
    <div
      aria-busy="false"
      aria-disabled="false"
      aria-expanded="false"
      aria-haspopup="true"
      aria-live="polite"
      aria-owns="rw_7_listbox"
      aria-readonly="false"
      class="sort-dropdown rw-dropdown-list rw-widget"
      id="rw_7_input"
      role="combobox"
      tabindex="0"
    >
      <div
        class="rw-widget-input rw-widget-picker rw-widget-container"
      >
        <div
          class="rw-input rw-dropdown-list-input"
        >
          <input
            class="rw-dropdown-list-autofill rw-detect-autofill rw-sr"
            tabindex="-1"
            value="Date added"
          />
          Date added
        </div>
        <span
          class="rw-select"
        >
          <button
            aria-disabled="false"
            aria-hidden="true"
            aria-label="open dropdown"
            class="rw-btn rw-btn-select"
            role="presentational"
            tabindex="-1"
            title="open dropdown"
            type="button"
          >
            <span
              aria-hidden="true"
              class="rw-i rw-i-caret-down"
            />
          </button>
        </span>
      </div>
    </div>
    <button
      class="sorting-toggle disabled"
      disabled=""
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-arrow-down fa-w-14 "
        data-icon="arrow-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M413.1 222.5l22.2 22.2c9.4 9.4 9.4 24.6 0 33.9L241 473c-9.4 9.4-24.6 9.4-33.9 0L12.7 278.6c-9.4-9.4-9.4-24.6 0-33.9l22.2-22.2c9.5-9.5 25-9.3 34.3.4L184 343.4V56c0-13.3 10.7-24 24-24h32c13.3 0 24 10.7 24 24v287.4l114.8-120.5c9.3-9.8 24.8-10 34.3-.4z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
